File: java/StreamingPipeInAndOut.java
import java.io.*;

public class StreamingPipeInAndOut {
    public static void main (String[] args) throws IOException {
        var inputPath = "streaming-in.pipe";
        var outputPath = "streaming-out.pipe";

        var input = new BufferedReader(new FileReader(inputPath));
        var output = new BufferedWriter(new FileWriter(outputPath));

        String line;
        while ((line = input.readLine()) != null) {
            output.write("received " + line + "\n");
            output.flush();
        }

        input.close();
        output.close();
    }
}

File: csharp/StreamingPipeInAndOut.cs
using System;
using System.IO;

class StreamingPipeInAndOut
{
    public static void Main(string[] args)
    {
        using var input = new StreamReader("streaming-in.pipe");
        using var output = new StreamWriter("streaming-out.pipe") { AutoFlush = true };

        string line;
        while ((line = input.ReadLine()) != null)
        {
            output.WriteLine($"received {line}");
        }
    }
}
